File: dotnet/cafdemalihapa/Applications/PathUtility.cs
namespace cafdemalihapa.Applications
{
    public static class PathUtility
    {
        public static string NormalizeSlashes(string path, SlashStyle style = SlashStyle.AutoByOS)
        {
            switch (style)
            {
                case SlashStyle.ForceSlash:
                    return path.Replace("\\", "/");
                case SlashStyle.ForceBackslash:
                    return path.Replace("/", "\\");
                case SlashStyle.AutoByOS:
                default:
                    char correctSlash = Path.DirectorySeparatorChar;
                    char wrongSlash = correctSlash == '\\' ? '/' : '\\';
                    return path.Replace(wrongSlash, correctSlash);
            }
        }

        public enum SlashStyle
        {
            AutoByOS,
            ForceSlash,
            ForceBackslash
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Applications/PathUtility.fs
namespace cafdemwimu.console.Applications

open System.IO

module PathUtility =

    type SlashStyle =
        | AutoByOS = 0
        | ForceSlash = 1
        | ForceBackslash = 2

    let NormalizeSlashes (path: string, style: SlashStyle) =
        match style with
        | SlashStyle.ForceSlash -> path.Replace("\\", "/")
        | SlashStyle.ForceBackslash -> path.Replace("/", "\\")
        | _ ->
            let correctSlash = Path.DirectorySeparatorChar
            let wrongSlash = if correctSlash = '\\' then '/' else '\\'
            path.Replace(wrongSlash, correctSlash)
